### ISSUE TITLE
Synapse createRoom 413 creates an Empty room

### ISSUE THREAD

**Issue by t3chguy** [matrix-org/synapse#8895]
If somehow your createRoom request is too large, like if for example you send a base64 encoded avatar instead of an mxc URL (wrongly) then the request fails with a 413 but a new Empty room is created and sent to the user with none of the parameters the user asked for.

Example request payload:

```json
{
  "visibility": "public",
  "name": "Test Space 17",
  "creation_content": {
    "type": "m.space"
  },
  "initial_state": [
    {
      "type": "m.room.avatar",
      "content": {
        "avatar": "data:image/png;base64,iVBOR..............AAAAASUVORK5CYII="
      }
    },
    {
      "type": "m.room.guest_access",
      "state_key": "",
      "content": {
        "guest_access": "can_join"
      }
    }
  ],
  "power_level_content_override": {
    "events_default": 100
  },
  "preset": "private_chat"
}
```

(avatar truncated Github would not let me post)

![image](https://user-images.githubusercontent.com/2403652/101487372-ffb53080-3955-11eb-889b-5e1868dd4b46.png)

Expectation: Empty room is not created.

Possibly related to https://github.com/matrix-org/synapse/issues/8625 but am operating in Monolith mode.

**Comment by richvdh** [comment]
yeah I think this is closely related to #8625: it boils down to the fact that `createRoom` is not atomic, and exceptions in the middle of it give you a broken room.

**Comment by richvdh** [comment]
Per #13427: the same happens if you hit the rate-limiter.

**Comment by RomanOlegovich** [comment]
@richvdh Can [workers](https://github.com/matrix-org/synapse/blob/develop/docs/workers.md)  help solve this problem?

**Comment by richvdh** [comment]
> @richvdh Can [workers](https://github.com/matrix-org/synapse/blob/develop/docs/workers.md) help solve this problem?

No, if anything they are likely to exacerbate the problem as it increases the chances of a failure creating one of the events in the new room.

**Comment by RomanOlegovich** [comment]
@richvdh What is a way to fix this and create room with expected alias, may be [delete](https://matrix-org.github.io/synapse/v1.39/admin_api/rooms.html#delete-room-api) broken room and re-create again?
